src/cyhy_cvesync/cve_sync.py

```diff
@@ -0,0 +1,171 @@
+
+            if cve_doc:  # Update existing CVE doc
+                if (
+                    cve_doc.cvss_score != cvss_base_score
+                    or cve_doc.cvss_version != cvss_version_temp
+                ):
+                    cve_doc.cvss_score = cvss_base_score
+                    cve_doc.cvss_version = cvss_version_temp
+                    await cve_doc.save()
+                    logger.info("Updated CVE document with id: %s", cve_id)
+                    updated_cve_docs_count += 1
+            else:  # Create new CVE doc
+                cve_doc = CVEDoc(
+                    id=cve_id,
+                    cvss_score=float(cvss_base_score),
+                    cvss_version=cvss_version_temp,
+                    severity=None,
+                )
+                await cve_doc.save()
+                logger.info("Created CVE document with id: %s", cve_id)
+                created_cve_docs_count += 1
+
+    return created_cve_docs_count, updated_cve_docs_count
+
+
+def fetch_cve_data(cve_url: str, gzipped: bool) -> dict:
+    """
+    Fetch the CVE data from the given URL.
+
+    This function retrieves Common Vulnerabilities and Exposures (CVE) JSON data from the specified URL.
+
+    Args:
+        cve_url (str): The URL to fetch the CVE JSON data from.
+        gzipped (bool): Whether the data is gzipped.
+
+    Returns:
+        dict: The CVE JSON data.
+
+    Raises:
+        urllib.error.HTTPError: If the CVE JSON cannot be retrieved.
+        ValueError: If the URL scheme is not allowed or if no data is received from the CVE URL.
+    """
+    # Create a Request object so we can test the safety of the URL
+    cve_request = urllib.request.Request(cve_url)
+    if cve_request.type not in ALLOWED_URL_SCHEMES:
+        raise ValueError("Invalid URL scheme in CVE JSON URL: %s" % cve_request.type)
+
+    # Below we disable the bandit blacklist for the urllib.request.urlopen() function
+    # since we are checking the URL scheme before using.
+
+    with urllib.request.urlopen(cve_url) as response:  # nosec B310
+        if response.status != 200:
+            raise urllib.error.HTTPError(
+                cve_url,
+                response.status,
+                "Failed to retrieve CVE data.",
+                response.headers,
+                None,
+            )
+
+        # Read the response content
+        response_content = response.read()
+        if not response_content:
+            raise ValueError("Empty response received from the server.")
+
+    if gzipped:
+        # Unzip the response content and return the JSON data
+        with gzip.GzipFile(fileobj=BytesIO(response_content)) as f:
+            return json.loads(f.read().decode("utf-8"))
+    else:
+        return json.loads(response_content)
+
+
+async def process_urls(
+    cve_urls: List[str], cve_data_gzipped: bool
+) -> Tuple[int, int, int]:
+    """Process URLs containing CVE data."""
```

Comment:
Return value documentation please.

Reply:
Please see https://github.com/cisagov/cyhy-cvesync/pull/1/commits/ce8375a815270fe3a06316ec8ab2f1637d72bca1 and resolve this conversation if that works for you.